### DISCUSSION TITLE
how to run command at build time?

### DISCUSSION THREAD

**Discussion by stsp** [mesonbuild/meson#13201] [Q&A]
`run_command()` is documented to work
on configure time. On build time there is
a `custom_target()`, but it wants to create
a file, rather than to return the command
output as a string.
Is there a similar function to `run_command()`,
but for build time?

**Comment by dreamer-coding** [comment]
Run target might do the trick: https://mesonbuild.com/Reference-manual_functions.html#run_target

**Reply by stsp** [reply]
I probably forgot to mention that I
need the stdout of the command I run.
With `run_command()` I can do
`capture: true`, which seems to be
not the case with `run_target()`?

**Reply by dreamer-coding** [reply]
I belave that can be done using `capture` from `run_command`. If set to `true`, any output generated on stdout will be captured and returned by the `.stdout()` method. If it is `false`, then `.stdout()` will return an empty string.

`custom_target` might be able to do the trick for this magic command https://mesonbuild.com/Reference-manual_functions.html#custom_target

**Reply by stsp** [reply]
In `custom_target()` it seems `capture:`
captures to the output file, not to `stdout()`.
How difficult is to add `capture:` to `run_target()`?

**Comment by stsp** [comment]
I filled #13206 ticket for this.
